### PR TITLE
Compute knownSize only once

### PR DESCRIPTION
**Pull request by allanrenucci**
This prevent possible exponential computations of knownSize

### CODE REVIEW

File: src/library/scala/collection/View.scala [hunk 1]
```diff
@@ -176,8 +176,10 @@ object View extends IterableFactory[View] {
   class Drop[A](underlying: SomeIterableOps[A], n: Int) extends AbstractView[A] {
     def iterator = underlying.iterator.drop(n)
     protected val normN = n max 0
-    override def knownSize =
-      if (underlying.knownSize >= 0) (underlying.knownSize - normN) max 0 else -1
+    override def knownSize = {
+      val size = underlying.knownSize
```
Comment (NthPortal) [inline src/library/scala/collection/View.scala:180]: my only concern is confusion between this local variable named `size` and the method named `size`

**Comment by dwijnand** [comment]
Is this necessary given `knownSize` is designed to be `O(1)`?

**Comment by allanrenucci** [comment]
> Is this necessary given knownSize is designed to be O(1)?

That's not always true. This is a follow up to scala/bug#11005

File: src/library/scala/collection/View.scala (hunk 1 above)
Comment (julienrf) [reply src/library/scala/collection/View.scala:180]: Ideas of possible names: `computedKnownSize`, `_knownSize`.

Comment (allanrenucci) [reply src/library/scala/collection/View.scala:180]: I think the name is fine. This is a local variable and the method is 2 lines of code. I don't believe there would be any confusion between a reference to the local variable and a call to the method `size`. Trying to come up with a fresh name would just make the method less readable

**Review by dwijnand — APPROVED** [review]
Alright then

**Pull request merged**